src/mapblock.cpp

```diff
@@ -27,13 +27,14 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "gamedef.h"
 #include "log.h"
 #include "nameidmapping.h"
-#include "content_mapnode.h" // For legacy name-id mapping
+#include "content_mapnode.h"  // For legacy name-id mapping
 #include "content_nodemeta.h" // For legacy deserialization
 #include "serialization.h"
 #ifndef SERVER
 #include "client/mapblock_mesh.h"
 #endif
 #include "porting.h"
+#include "scripting_server.h"
```

Comment:
importing server scripting into the mapblock definitely feels wrong
can't we leave this in `ServerEnvironment`?

Reply:
i totally agree that it's not convenient as mapblock is currently generic and we have no supersede class on Server and Client to handle those types of changes. Let's see if a lamdba generic callback can fit

Reply:
ok i used a callback function to perform this action which make generic triggering on node timer, if we need it client side it will be re-usable :)